shop/src/main/java/ro/msg/learning/shop/controller/impl/ProductControllerImpl.java

```diff
@@ -0,0 +1,70 @@
+package ro.msg.learning.shop.controller.impl;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.controller.ProductController;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.dto.ProductToSaveDto;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.utils.mapper.ProductMapper;
+
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/products")
+public class ProductControllerImpl implements ProductController {
+    private static final String ID = "id";
+    private static final String CREATE_SUCCESS = "Product successfully created";
+    private static final String DELETE_SUCCESS = "Product successfully deleted";
+    private static final String UPDATE_SUCCESS = "Product successfully updated";
+    private final ProductService productService;
+
+    public ProductControllerImpl(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping
+    @Override
+    public List<ProductDto> findAllProducts() {
+        return productService.findAllProducts().stream()
+                .map(ProductMapper::ProductToDto)
```

Comment:
The controller needs to be a **dumb component**. It should only know authentication (in the future) and who to pass the responsability to. The service is the layer which should contain all the business logic including such mappings